baseplate/frameworks/queue_consumer/kafka.py

```diff
@@ -235,17 +298,22 @@ def new(
         :param kafka_consume_batch_size: The number of messages the `KafkaConsumerWorker`
             reads from Kafka in each batch. Defaults to 1.
         :param message_unpack_fn: A function that takes one argument, the `bytes` message body
-            and returns the message in the format the handler expects. Defaults to `json.loads`.
+            and returns the message in the format the handler expects. If no input is provided,
+            a function is automatically picked one based on whether a schema_registry was provided or not.
         :param health_check_fn: A `baseplate.server.queue_consumer.HealthcheckCallback`
             function that can be used to customize your health check.
+        :param schema_registry: Schema Registry where the schemas of the Avro messages are stored
 
         """
 
-        service_name, _, group_name = group_id.partition(".")
-        assert service_name and group_name, "group_id must start with 'SERVICENAME.'"
-        assert name == f"kafka_consumer.{group_name}"
+        validate_group_id(name, group_id)
+        consumer = make_kafka_consumer(bootstrap_servers, group_id, topics, schema_registry)
 
-        consumer = cls.make_kafka_consumer(bootstrap_servers, group_id, topics)
+        if message_unpack_fn is None:
```

Comment:
Do you want to raise an exception if both `message_unpack_fn` and `schema_registry` are supplied?

Reply:
this confused me too, but we could have a case where we actually intended to specify both, like if we're using avro and set `message_unpack_fn=lambda m: consumer._serializer.decode_message(m).bar`